operation/fhir-operation-validate/src/main/java/org/linuxforhealth/fhir/operation/validate/ValidateOperation.java

```diff
@@ -131,20 +120,88 @@ private OperationOutcome buildResourceValidOperationOutcome(List<Issue> issues)
     
     
     /**
-     * Check if a resource validation mode code is valid.
+     * This method does the following validations.
+     * 1. Validate if a resource validation mode code is valid.
+     * 2. Validate an interaction for a specified resource type.
+     * 3. Validate if persistence layer implementation supports update/create mode if mode = create.
+     * 4. Validate if persistence layer implementation supports update/create mode if mode = update but the resource doesnot exist yet in the DB.
+     * 5. Validate if modes update and delete are only be used when the operation is invoked at the resource instance level
      *
      * @param modeParameter resource validation mode code to be validated
-     * @throws FHIROperationException when the resource validation mode code is invalid
+     * @param resourceHelper Resource operation provider for loading related Library resources
+     * @param resourceType a valid resource type string
+     * @param operationContext the FHIROperationContext associated with the request
+     * @throws Exception 
+     */
+    private void validateModeParameter(ModeType modetype, FHIRResourceHelpers resourceHelper, String resourceType, FHIROperationContext operationContext, String logicalId) throws Exception {
+        if(modetype == null) {
```

Comment:
```
        Objects.requireNonNull(modeType, "modelType");
```

Reply:
I'm assuming that if validateModeParameter is called with a null value that is an error. The javadoc doesn't say if null is allowed.

Reply:
I'm OK either way.

Reply:
```suggestion
        if (modetype == null) {
```